### ISSUE TITLE
Focus is not working

### ISSUE THREAD

**Issue by Helderjfl** [SuslikV/cfg-cam#4]
For me, this app works for everything, except for focus. It does change to manual, but the value is not actually setted. I don't know if its something with the webcam or the app.

My cam_sett
```
/ WebCameraConfig settings file
Device #2
\\?\usb#vid_046d&pid_085b&mi_00#6&1663cfd6&0&0000#{65e8773d-8f56-11d0-a3b9-00a0c9223196}\global
Logitech Webcam C925e
VideoProcAmp_BacklightCompensation=1 [Manual]
VideoProcAmp_Brightness=135 [Manual]
VideoProcAmp_Contrast=130 [Manual]
VideoProcAmp_Gain=0 [Manual]
VideoProcAmp_Saturation=130 [Manual]
VideoProcAmp_Sharpness=135 [Manual]
VideoProcAmp_WhiteBalance=5330 [Auto]
CameraControl_Exposure=-5 [Auto]
CameraControl_Focus=10 [Manual]
CameraControl_Pan=0 [Manual]
CameraControl_Tilt=0 [Manual]
CameraControl_Zoom=100 [Manual]
---end of the device #2
```

**Comment by dancariola** [comment]
Hi Helderjfl - I have the Logitech WebCam C922  and mine successfully saves all of the above settings and reapplies them on execution.

Device #2
\\?\usb#vid_046d&pid_085c&mi_00#6&340648ad&0&0000#{65e8773d-8f56-11d0-a3b9-00a0c9223196}\global
C922 Pro Stream Webcam
VideoProcAmp_BacklightCompensation=1 [Manual]
VideoProcAmp_Brightness=150 [Manual]
VideoProcAmp_Contrast=128 [Manual]
VideoProcAmp_Gain=0 [Manual]
VideoProcAmp_Saturation=128 [Manual]
VideoProcAmp_Sharpness=151 [Manual]
VideoProcAmp_WhiteBalance=4000 [Auto]
CameraControl_Exposure=-5 [Auto]
CameraControl_Focus=20 [Manual]
CameraControl_Pan=0 [Manual]
CameraControl_Tilt=0 [Manual]
CameraControl_Zoom=130 [Manual]
---end of the device #2

**Comment by SuslikV** [comment]
@Helderjfl unfortunately, I don't have an example of this hardware, so cannot say what may gonna wrong. Maybe camera is too smart and this part (focus) somehow controlled through the driver. Then this out of scope for this application. Sorry. You need to look for other util.

**Comment by Helderjfl** [comment]
Yeah, I don't know exactly what it is. Every other setting is applied. I tried Logitech Camera Settings, which seems to be the only Logitech Software that supports it, and when I open it the focus is applied, even though cfg-cam defined the value.

**Comment by Vallek** [comment]
Same problem here. Except I have Canyon CNS-CWC5 so i can't use logitech software. 

It also turns auto focus off but doesn't really apply number and just leaves 0.

Anyway thanks for this tool. Still better than nothing)

PS Funny thing is if I manually start this tool again it applies correct focus.

**Comment by Feldheld** [comment]
I had the same issue with focus at first but then with another focus setting it suddenly worked. 

Even in the DirectShow dialog it doesnt save properly. I can move the slider to 27 focus and save it but when I open the dialog again its back the the former focus setting. It seems that it can only save increments of 5, i.e. 0, 5, 10, 15 etc and it simply ignores every other number. 

edit: and kudos to SuslikV for providing this nice little tool!

**Comment by iohann95** [comment]
> Same problem here. Except I have Canyon CNS-CWC5 so i can't use logitech software.
> 
> It also turns auto focus off but doesn't really apply number and just leaves 0.
> 
> Anyway thanks for this tool. Still better than nothing)
> 
> PS Funny thing is if I manually start this tool again it applies correct focus.

Hey, that also happens with me.
You know what's funny, if I restart the computer, it does work the first time I open the program. But if I just turned on the computer after a hard power on, I need to open it twice.
